document.c

```diff
@@ -3397,51 +3456,38 @@ static size_t
 html_find_end(const char *tag, size_t tag_len,
 	struct lowdown_doc *doc, const char *data, size_t size)
 {
-	size_t	i, w = 0;
+	size_t i, j, w = 0, depth = 0;
+	const char *matched_tag;
 
 	for (i = 0; ; i++) {
+		/* Find next opening tag */
 		while (i < size && data[i] != '<')
 			i++;
+		/* Ran out of data to check */
 		if (i >= size)
 			return 0;
-		w = html_is_end(tag, tag_len, doc, data + i, size - i);
-		if (w)
-			break;
-	}
-
-	return i + w;
-}
-
-/*
- * Try to find end of HTML block in strict mode (it must be an
- * unindented line, and have a blank line afterwards).
- * Returns the length on match, 0 otherwise.
- */
-static size_t
-html_find_end_strict(const char *tag, size_t tag_len,
```

Comment:
Removed `html_find_end_strict` as the way it was coded did not allow enough context to know about nested HTML blocks that may have the same tag type (such as nested divs).